src/lib/importer/hoyoLabFormatParser.tsx

```diff
@@ -0,0 +1,205 @@
+      lightConeSuperimposition: 1,
+    }
+    if (character.equip != null) {
+      characterData.lightCone = character.equip.id.toString()
+      characterData.lightConeSuperimposition = character.equip.rank
+    }
+    output.characters.push(characterData)
+    for (const relic of [...character.relics, ...character.ornaments]) {
+      const substats: {
+        stat: SubStats
+        value: number
+        addedRolls: number
+      }[] = []
+      for (const property of relic.properties) {
+        const substat = {
+          stat: getStat(property.property_type),
+          value: readValue(property.value),
+          addedRolls: property.times - 1,
+        }
+        substats.push(substat)
+      }
+      output.relics.push({
+        enhance: relic.level,
+        equippedBy: (character.id).toString(),
+        grade: relic.rarity,
+        id: uuidv4(),
+        part: getSlot(relic.pos),
+        set: getSet(relic.id),
+        main: {
+          stat: getStat(relic.main_property.property_type),
+          value: readValue(relic.main_property.value),
+        },
+        substats: substats,
+        verified: false,
+      })
+    }
+    if (character.id == 8002) {
+      output.metadata.trailblazer = 'Stelle'
+      output.metadata.current_trailblazer_path = 'Destruction'
+    }
+    if (character.id == 8004) {
+      output.metadata.trailblazer = 'Stelle'
+      output.metadata.current_trailblazer_path = 'Preservation'
+    }
+    if (character.id == 8006) {
+      output.metadata.trailblazer = 'Stelle'
+      output.metadata.current_trailblazer_path = 'Harmony'
+    }
+    if (character.id == 8001) {
+      output.metadata.trailblazer = 'Caelus'
+      output.metadata.current_trailblazer_path = 'Destruction'
+    }
+    if (character.id == 8003) {
+      output.metadata.trailblazer = 'Caelus'
+      output.metadata.current_trailblazer_path = 'Preservation'
+    }
+    if (character.id == 8005) {
+      output.metadata.trailblazer = 'Caelus'
+      output.metadata.current_trailblazer_path = 'Harmony'
+    }
+  }
+  output.relics.map((r) => RelicAugmenter.augment(r))
+  return output
+}
+
+function readValue(value: string) {
+  if (value.endsWith('%')) {
+    return Utils.precisionRound(parseFloat(value.slice(0, value.length - 1)))
+  }
+  return Utils.precisionRound(parseFloat(value))
+}
+
+function getStat(id: number) {
+  switch (id) {
+    case 27:
+      return Constants.Stats.HP
+    case 32:
+      return Constants.Stats.HP_P
+    case 29:
+      return Constants.Stats.ATK
+    case 33:
+      return Constants.Stats.ATK_P
+    case 31:
+      return Constants.Stats.DEF
+    case 34:
+      return Constants.Stats.DEF_P
+    case 51:
+      return Constants.Stats.SPD
+    case 52:
+      return Constants.Stats.CR
+    case 53:
+      return Constants.Stats.CD
+    case 56:
+      return Constants.Stats.EHR
+    case 57:
+      return Constants.Stats.RES
+    case 59:
+      return Constants.Stats.BE
+    case 54:
```

Comment:
Object lookups are easier to maintain/read/catch errors than switch cases, refactoring to:

```

type TrailblazerMetadata = { trailblazer: string; current_trailblazer_path: string }
const trailblazerMetadataLookup: { [key: number]: TrailblazerMetadata } = {
  8001: { trailblazer: 'Caelus', current_trailblazer_path: 'Destruction' },
  8002: { trailblazer: 'Stelle', current_trailblazer_path: 'Destruction' },

  8003: { trailblazer: 'Caelus', current_trailblazer_path: 'Preservation' },
  8004: { trailblazer: 'Stelle', current_trailblazer_path: 'Preservation' },

  8006: { trailblazer: 'Stelle', current_trailblazer_path: 'Harmony' },
  8005: { trailblazer: 'Caelus', current_trailblazer_path: 'Harmony' },
}
function getTrailblazerMetadata(id: number) {
  return trailblazerMetadataLookup[id] || trailblazerMetadataLookup[8002]
}

const statLookup: { [key: number]: string } = {
  27: Constants.Stats.HP,
  32: Constants.Stats.HP_P,
  29: Constants.Stats.ATK,
  33: Constants.Stats.ATK_P,
  31: Constants.Stats.DEF,
  34: Constants.Stats.DEF_P,
  51: Constants.Stats.SPD,
  52: Constants.Stats.CR,
  53: Constants.Stats.CD,
  56: Constants.Stats.EHR,
  57: Constants.Stats.RES,
  59: Constants.Stats.BE,
  54: Constants.Stats.ERR,
  55: Constants.Stats.OHB,
  12: Constants.Stats.Physical_DMG,
  14: Constants.Stats.Fire_DMG,
  16: Constants.Stats.Ice_DMG,
  18: Constants.Stats.Lightning_DMG,
  20: Constants.Stats.Wind_DMG,
  22: Constants.Stats.Quantum_DMG,
  24: Constants.Stats.Imaginary_DMG,
}
function getStat(id: number) {
  return statLookup[id] || ''
}

const slotLookup: { [key: number]: string } = {
  1: Parts.Head,
  2: Parts.Hands,
  3: Parts.Body,
  4: Parts.Feet,
  5: Parts.PlanarSphere,
  6: Parts.LinkRope,
}
function getSlot(id: number) {
  return slotLookup[id] || ''
}
```